### PR TITLE
[Tailwind] Adds Remix example

### PR DESCRIPTION
**Pull request by QuintonC**
## ℹ️ What is the context for these changes?

Adds example for the Remix framework.

## 🕹️ Demonstration

Screenshot 2023-03-16 at 11 58 41 AM

## 🎩 How can this be tophatted?

1. Pull this branch
2. Ensure you've run `yarn install` followed by `yarn build:packages` at the repository root directory
3. Navigate to the Remix example application
4. Run `yarn install` in the Remix example directory
5. Run `yarn dev`

### CODE REVIEW

File: examples/remix/remix.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/** @type {import('@remix-run/dev').AppConfig} */
+module.exports = {
+  ignoredRouteFiles: ['**/.*'],
+  serverDependenciesToBundle: ['@shopify/connect-wallet', /^@?wagmi.*/, /.*/],
```
Comment (QuintonC) [inline examples/remix/remix.config.js:4]: I was running into strange issues with using wagmi and found this snippet in the Rainbow examples.

File: examples/remix/app/root.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    <html lang="en">
+      <head>
+        <Meta />
+        <Links />
+        {typeof document === 'undefined' ? '__STYLES__' : null}
+      </head>
+      <body>
+        <WagmiConfig client={client}>
+          <ConnectWalletProvider chains={chains} connectors={[]}>
```
Comment (lesliexin) [inline examples/remix/app/root.tsx:39]:
```suggestion
          <ConnectWalletProvider chains={chains} connectors={connectors}>
```

File: examples/remix/app/root.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  Meta,
+  Outlet,
+  Scripts,
+  ScrollRestoration,
+} from '@remix-run/react';
+import {ConnectWalletProvider} from '@shopify/connect-wallet';
+import stylesheet from '@shopify/connect-wallet/styles.css';
+import {WagmiConfig} from 'wagmi';
+
+import {chains, client} from '~/connect-wallet-config';
```
Comment (lesliexin) [inline examples/remix/app/root.tsx:14]:
```suggestion
import {chains, client, connectors} from '~/connect-wallet-config';
```

File: examples/remix/app/connect-wallet-config.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+const {chains, provider, webSocketProvider} = configureChains(
+  [mainnet],
+  [
+    // alchemyProvider({apiKey: 'INSERT API KEY HERE'}),
+    publicProvider(),
+  ],
+);
+
+const {connectors} = getDefaultConnectors({chains});
```
[the remark this answers is not in the document]
Comment (lesliexin) [reply examples/remix/app/connect-wallet-config.ts:15]: (PR for updating the other examples [here](https://github.com/Shopify/blockchain-components/pull/132))

File: examples/remix/app/root.tsx (hunk 3 above)
Comment (QuintonC) [reply examples/remix/app/root.tsx]: Updated in [c5fbb5d](https://github.com/Shopify/blockchain-components/commit/c5fbb5d977b2678f230af5b56fb66ccf4d32e1a3)

File: examples/remix/package.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{
+  "private": true,
+  "sideEffects": false,
+  "scripts": {
```
Comment (lesliexin) [inline examples/remix/package.json:4]: Do we need the `bundle-packages` script? ([Vite](https://github.com/Shopify/blockchain-components/blob/main/examples/vite/package.json#L7))

Comment (QuintonC) [reply examples/remix/package.json:4]: > Do we need the `bundle-packages` script? ([Vite](https://github.com/Shopify/blockchain-components/blob/main/examples/vite/package.json#L7))

I think there is actually an issue with the bundle packages script that I didn't realize until today.

There is a bunch of strange conflicts with running the build prior to running the development server. The bundled packages don't update the code in the example projects' `node_modules`.

I'm probably going to address this as a follow up after the tailwind migration. My thought is that we might need to make use of concurrently (seen in the package.json for `connect-wallet` and `tokengate` packages) and run the dev server alongside the dev command for the packages.

**Pull request merged**